File: python/71.simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:

        arr = path.split("/")
        out = ""

        for sub in arr:
            if sub == ".." and len(out) > 0:
                out = out[:out.rindex('/')]
            elif sub not in (".","",".."):
                out += ("/" + sub)

        return out if len(out) > 0 else "/"

File: go/71.simplify-path.go
func simplifyPath(path string) string {
    dirs := strings.Split(path, "/")
    canonicalPath := make([]string, 0, len(dirs))
    for _, d := range dirs {
        if d == ".." {
            if len(canonicalPath) > 0 {
                canonicalPath = canonicalPath[:len(canonicalPath)-1]
            }
        } else if d != "." && d != "" {
            canonicalPath = append(canonicalPath, d)
        }
    }
    return "/" + strings.Join(canonicalPath, "/")
}